### DISCUSSION TITLE
TypeScript ignores query condition and reports error.

### DISCUSSION THREAD

**Discussion by jeton-th** [TanStack/query#3795] [Q&A] [answered]
TypeScript throws an error about the parameter being potentially undefined.

![image](https://user-images.githubusercontent.com/5147537/178190514-ed8bb435-7af0-4b65-88e2-cf0bc9ae6c3d.png)

How do I tell TypeScript that the query will run conditionally, only if the parameter exists?

**Comment by TkDodo** [comment, marked as answer]
https://tkdodo.eu/blog/react-query-and-type-script#type-safety-with-the-enabled-option
